### PR TITLE
2021 CSS chapter

### PR DESCRIPTION
**Pull request by meyerweb**
Progress on #2140 

Also corrected some errors in figure construction around image weight and counts.  I hope I didn’t introduce more errors!

Staged URL: https://20211115t173622-dot-webalmanac.uk.r.appspot.com/en/2021/css

### CODE REVIEW

File: src/content/en/2021/css.md [hunk 1]
```diff
@@ -590,28 +672,39 @@ featured_stat_label_3: TODO
...
+
+
+### Number of images in CSS
+
+What we found was that most CSS doesn’t result in a lot of image loads: the lower two percentiles came in at one image each, and even the 90th percentile hovered around 10 images, across all image types.
 
 {{ figure_markup(
   image="number-of-images-loaded.png",
-  caption="Distribution of the number of external images loaded via CSS.",
+  caption="Distribution of the median number of external images loaded via CSS.",
```
Comment (rviscomi) [inline src/content/en/2021/css.md:686]: Could you explain your reasoning for including the word "median" here? The distribution is summarized by the percentiles, one of which is the median, but I wouldn't say that it's a "distribution of the median".
```suggestion
  caption="Distribution of the number of external images loaded via CSS.",
```

File: src/content/en/2021/css.md [hunk 2]
```diff
@@ -754,18 +864,29 @@ featured_stat_label_3: TODO
...
+* Hard color stops — that is, gradients where two color stops were placed at the same position — occurred in just over half of all gradients.
+* Color-stop interpolation (a.k.a “midpoints”) were used in 21% of all gradient instances.
+
+We also saw a dramatic reduction at the top end of gradient complexity.  Last year, the gradient with the largest number of color stops had 646 stops.  This year, the winner had only 81 color stops.
 
 {{ figure_markup(
   image="most-color-stops.png",
   classes="height-16vw-122px",
-  caption="The linear gradient with the most color stops.",
+  caption="The linear gradient with the most color stops foudn in our analysis.",
```
Comment (rviscomi) [inline src/content/en/2021/css.md:880]: Removing as it's redundant to disclaim that the results are what we found in our analysis. That could apply to every stat in the chapter! :)

```suggestion
  caption="The linear gradient with the most color stops.",
```

File: src/content/en/2021/css.md [hunk 3]
```diff
@@ -743,7 +853,7 @@ featured_stat_label_3: TODO
   ) }}</figcaption>
 </figure>
 
-
+Linear gradients continue to be the clear favorite, maintaining the 5-to-1 lead over radial gradients seen in the 2020 Web Almanac.
```
Comment (rviscomi) [inline src/content/en/2021/css.md:856]: Is there a specific section you wanted to deep-link to?
```suggestion
Linear gradients continue to be the clear favorite, maintaining the 5-to-1 lead over radial gradients seen in the [2020 Web Almanac](../2020/css).
```

**Review by rviscomi — CHANGES_REQUESTED** [review]

File: src/content/en/2021/css.md [hunk 4]
```diff
@@ -1087,6 +1360,14 @@ featured_stat_label_3: TODO
...
+
+The other low scoring property, `font-stretch`, is heavily dependent on both font families having wide or narrow faces available and authors choosing (or knowing) to make use of them, so its 5% share (down from 8% last year) comes as little surprise.
+
+
+#### Flexbox
+
+Some of the Flexbox longhand and shorthand properties have had a turbulent history; for example, the CSS Flexbox specification itself [recommends that authors avoid ](https://drafts.csswg.org/css-flexbox-1/#flex-grow-property)using `flex-grow`, `flex-shrink`, and `flex-basis` and use the `flex` shorthand instead.  This ensures that unset properties have sensible values. Unfortunately, this doesn’t seem to be bearing out in the wild, where `flex-basis` is used more often on mobile pages than is `flex`, by a margin of more than 10%.
```
Comment (rviscomi) [inline src/content/en/2021/css.md:1370]: This resolves the linter error

```suggestion
Some of the Flexbox longhand and shorthand properties have had a turbulent history; for example, the CSS Flexbox specification itself [recommends that authors avoid](https://drafts.csswg.org/css-flexbox-1/#flex-grow-property) using `flex-grow`, `flex-shrink`, and `flex-basis` and use the `flex` shorthand instead.  This ensures that unset properties have sensible values. Unfortunately, this doesn’t seem to be bearing out in the wild, where `flex-basis` is used more often on mobile pages than is `flex`, by a margin of more than 10%.
```

**Review by rviscomi — CHANGES_REQUESTED** [review]

File: src/content/en/2021/css.md [hunk 5]
```diff
@@ -17,9 +17,22 @@ featured_stat_3: TODO
 featured_stat_label_3: TODO
 ---
```
Comment (rviscomi) [inline src/content/en/2021/css.md:18]: @meyerweb could you also start filling in the metadata at the top of the file?

Comment (meyerweb) [reply src/content/en/2021/css.md:18]: Yep, will do.

File: src/content/en/2021/css.md [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
 #See https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters
 title: CSS
-description: TODO
-authors: []
-reviewers: []
-analysts: []
+description: CSS chapter of the 2021 Web Almanac covering trends, changes, and patterns in CSS use across the web.
+authors: [meyerweb, GeekBoySupreme]
+reviewers: [svgeesus, j9t, estelle, bkardell, argyleink, LeaVerou]
+analysts: [rviscomi, thecraftysoul]
```
Comment (rviscomi) [inline src/content/en/2021/css.md:7]: I don't recall thecraftysoul contributing, do you?

```suggestion
analysts: [rviscomi]
```

Comment (meyerweb) [reply src/content/en/2021/css.md:7]: I don’t.  I thought maybe they’d done back-of-house work with you, and I didn’t want to leave anyone out.

**Comment by meyerweb** [comment]
This PR says there’s a change request to be dealt with, but I don’t see where to resolve it.  Point me in the right direction and I’ll clear whatever blocker I can!

**Comment by tunetheweb** [comment]
Let’s get #2480 updated and merged first and then can do one final merge conflict resolution here while we work through the chapter (which will take longer than just adding the contributors - hence why that’s in a separate PR in #2480)

**Comment by rviscomi** [comment]
Ah I had just fixed the merge conflict. I'm ok to close #2480 and merge this PR to proceed to the editing stage.

**Review by rviscomi — APPROVED** [review]

**Comment by tunetheweb** [comment]
> Ah I had just fixed the merge conflict. I'm ok to close #2480 and merge this PR to proceed to the editing stage.

Wow didn't realise it was that close! Awesome!!!

**Pull request merged**